src/main/java/com/browserstack/WebDriverRuntime.java

```diff
@@ -0,0 +1,190 @@
+    private final int limit;
+    private final FeatureSupplier featureSupplier;
+    private final PickleOrder pickleOrder;
+    private final CucumberExecutionContext context;
+    private final BatchExecutionRunner batchExecutionRunner;
+    private final boolean isRerunEnabled;
+
+    private WebDriverRuntime(EventBus eventBus, CucumberExecutionContext context, Predicate<Pickle> filter, int limit, FeatureSupplier featureSupplier, PickleOrder pickleOrder, BatchExecutionRunner batchExecutionRunner, boolean isRerunEnabled, RuntimeOptions runtimeOptions) {
+        this.eventBus = eventBus;
+        this.context = context;
+        this.filter = filter;
+        this.limit = limit;
+        this.featureSupplier = featureSupplier;
+        this.pickleOrder = pickleOrder;
+        this.batchExecutionRunner = batchExecutionRunner;
+        this.isRerunEnabled = isRerunEnabled;
+        eventBus.send(new RuntimeCreated(Instant.now(), runtimeOptions));
+    }
+
+    public static WebDriverRuntime.Builder builder() {
+        return new WebDriverRuntime.Builder();
+    }
+
+    public void run() {
+
+        eventBus.send(new BuildStarted(Instant.now()));
+
+        // Feature Scanning
+        this.context.startTestRun();
+        this.execute(() -> {
+            this.context.runBeforeAllHooks();
+            this.runFeatures();
+        });
+
+        // Context closure for single execution
+        if (!isRerunEnabled) {
+            batchExecutionRunner.closeExecutionContext();
+        }
+
+    }
+
+    private void runFeatures(){
+        List<Feature> features = this.featureSupplier.get();
+        CucumberExecutionContext cucumberExecutionContext = this.context;
+        Objects.requireNonNull(cucumberExecutionContext);
+        features.forEach(cucumberExecutionContext::beforeFeature);
+        List<Execution> executions = new ArrayList<>();
+        features.stream()
+                .flatMap((feature) -> {
+                    List<Pickle> pickles = feature.getPickles().stream()
+                            .filter(this.filter)
+                            .collect(Collectors.collectingAndThen(Collectors.toList(), (list) -> this.pickleOrder.orderPickles(list)))
+                            .stream().limit(this.limit > 0 ? (long) this.limit : 2147483647L)
+                            .collect(Collectors.toList());
+                    return pickles.stream().map(pickle -> new PickleFeature(pickle, feature));
+                })
+                .forEach(pickleFeature -> webDriverFactory.getPlatforms().forEach(platform -> executions.add(new Execution(platform, pickleFeature.getFeature(), pickleFeature.getPickle()))));
+        // Pickle execution
+        batchExecutionRunner.submitExecutions(0, executions);
+    }
+
+
+    private void execute(Runnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable var3) {
+            UnrecoverableExceptions.rethrowIfUnrecoverable(var3);
```

Comment:
let's either use default suggested name for object OR a meaningful content-based name for the object instead of var3